apps/hannk/delegate/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# --------------- registrar (static-library) delegate
+
+add_library(hannk_delegate_registrar STATIC
+            hannk_delegate_provider.cpp
+            hannk_delegate.cpp)
+target_link_libraries(hannk_delegate_registrar PRIVATE
+                      interpreter
+                      error_util
+                      hannk_log_tflite
+                      tensorflowlite_headers)
+target_include_directories(hannk_delegate_registrar PRIVATE
+                           $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
+                           $<BUILD_INTERFACE:${CMAKE_PREFIX_PATH}/include>)
+
+
+# --------------- external (shared-library) delegate
+
+# External delegates in TFLite are always .so (never .dylib);
+# on Mac we must declare the library to be MODULE rather than SHARED to get this
+if (APPLE)
+set(SOTYPE MODULE)
+else ()
+set(SOTYPE SHARED)
+endif ()
+
+add_library(hannk_delegate_external ${SOTYPE}
+            hannk_delegate_adaptor.cpp
+            hannk_delegate.cpp)
+target_link_libraries(hannk_delegate_external PRIVATE
+                      interpreter
+                      error_util
+                      hannk_log_tflite
+                      tensorflowlite_headers)
+target_include_directories(hannk_delegate_external PRIVATE
+                           $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
+                           $<BUILD_INTERFACE:${CMAKE_PREFIX_PATH}/include>)
+if (APPLE)
+    target_link_options(hannk_delegate_external PRIVATE "-Wl,-exported_symbols_list,${CMAKE_CURRENT_SOURCE_DIR}/exported_symbols.osx")
+else ()
+    target_link_options(hannk_delegate_external PRIVATE "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/exported_symbols.ldscript"
+                                                      "-Wl,-soname,libHannkDelegate.so")
```

Comment:
Why is the `-soname` flag necessary when `OUTPUT_NAME` is set below?

Reply:
Also, does Hannk not support Windows? If so, that should be checked in the top level. If not, this needs to be adjusted to not add `exported_symbols.ldscript` on Windows.